qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -208,29 +356,21 @@ def get_solver(  # type: ignore[override]
         if initial_state is None:
             initial_state = HartreeFock(num_spin_orbitals, num_particles, qubit_converter)
 
-        ansatz = self.ansatz
-        if ansatz is None:
-            ansatz = UCCSD()
-        ansatz.qubit_converter = qubit_converter
-        ansatz.num_particles = num_particles
-        ansatz.num_spin_orbitals = num_spin_orbitals
-        ansatz.initial_state = initial_state
-
-        # TODO: leverage re-usability of VQE after fixing
-        # https://github.com/Qiskit/qiskit-terra/issues/7093
-        vqe = VQE(
-            ansatz=ansatz,
-            quantum_instance=self.quantum_instance,
-            optimizer=self.optimizer,
-            initial_point=self.initial_point,
-            gradient=self.gradient,
-            expectation=self.expectation,
-            include_custom=self.include_custom,
-            callback=self.callback,
-            **self._kwargs,
-        )
-
-        return vqe
+        if self.ansatz is None:
+            self._vqe.ansatz = UCCSD()
+        self._vqe.ansatz.qubit_converter = qubit_converter
+        self._vqe.ansatz.num_particles = num_particles
+        self._vqe.ansatz.num_spin_orbitals = num_spin_orbitals
+        self._vqe.ansatz.initial_state = initial_state
+        # maintain initialization done in vqe constructor
+        logger.info(self._vqe.print_settings())
```

Comment:
This line of code is in VQE constructor and does "magic" :). 
The main issue here is that the ansatz has all the members optional in constructor and is initialised whenever the caller wants to. So, if we just set an ansatz, we may have an object which is invalid. If it is fully set, still the number of qubits are not calculated properly.  This line `_vqe.print_settings()` calls` self.ansatz.draw(output="text")` which builds the circuit, i.e. has the right members set. This is not clean and relates to the comment above (we may add an invalid ansatz to VQE and it fails when logging the information...).

Reply:
Ok, I would have to look. The print_settings() in VQE is something we had in the very early days such that you could see all those aspects when running it if you wanted. Quite a bit has changed since then. But I will note that the heuristic anstazes likes TwoLocal etc are BluePrint circuits and also are in an incomplete state (trying not to say invalid here) until the number of qubits is set when the operator is finally given to VQE. I.e they are more like template - it knows how to build but it needs to be told how big - num_qubits, Until then its just an emply template if you will. Now with UCC I guess the num qubits is more indirectly set, but given None is a valid value for num particles etc it ought to support draw in this state without failing.

Reply:
@woodsp-ibm, thank you for looking into the issue!
Yes, there is a side effect. If we remove that log/print_settings, some tests will fail as the number of qubits in the ansatz( 0) doesn't match the number of qubits needed in the circuit. So, the ansatz is not "built" just by setting the properties.

Reply:
Update: I have UCC working locally such that all the current unit tests pass without the side-effect from the logging call in vqe (presently I deleted the logging call in VQE though I may leave it but change to have it inside an if that checks if info level logging is enabled.). I want to reflect on the change over the weekend, but expect to do a PR early next week.

Reply:
@iuliazidaru I made a change to UCC in #601, which is now merged. You should no longer need to rely on the side-effect of that logging statement for things to work. The equivalent change still needs to be done to UVCC - see #602 - but I think things should now be fine for the VQE UCC factory.

Reply:
UVCC is also now fixed (by #615)